packages/account/src/providers/provider.ts

```diff
@@ -1677,15 +1677,17 @@ export default class Provider {
    */
   async getBlockWithTransactions(
     /** ID or height of the block */
-    idOrHeight: string | number | 'latest'
+    idOrHeight: BigNumberish | 'latest'
   ): Promise<(Block & { transactions: Transaction[] }) | null> {
     let variables;
     if (typeof idOrHeight === 'number') {
       variables = { blockHeight: bn(idOrHeight).toString(10) };
     } else if (idOrHeight === 'latest') {
       variables = { blockHeight: (await this.getBlockNumber()).toString() };
-    } else {
+    } else if (typeof idOrHeight === 'string' && idOrHeight.length === 66) {
```

Comment:
```suggestion
    } else if (isB256(idOrHeight)) {
```

Is this checking for `b256`? May be a bit more verbose to use this util, or add a doc?